misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.maven;
+
+import org.ballerinalang.maven.exceptions.MavenResolverException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit Tests for {@link MavenResolver}.
+ *
+ *  @since 2.0.0
+ */
+public class MavenResolverUnitTest {
+    private MavenResolver resolver = mock(MavenResolver.class);
```

Comment:
What is the class we are trying to test here? Isn't it the `MavenResolver`?

Reply:
Yes. This is to test the functions of MavenResolverClass.

Reply:
Then we shouldn't be mocking the same class. Mocking is used to mimic other classes, not the class under test.